axum-extra/src/protobuf.rs

```diff
@@ -0,0 +1,303 @@
+//! Protocol Buffer extractor and response.
+
+use axum::{
+    async_trait,
+    body::{Bytes, HttpBody},
+    extract::{rejection::BytesRejection, FromRequest, RequestParts},
+    response::{IntoResponse, Response},
+    BoxError,
+};
+use http::StatusCode;
+use prost::Message;
+use std::ops::{Deref, DerefMut};
+
+/// Protocol Buffer extractor and response.
+///
+/// When used as an extractor, it can decode request bodies into some type that
+/// implements [`prost::Message`]. The request will be rejected (and a [`ProtoBufRejection`] will
+/// be returned) if:
+///
+/// - The body couldn't be decoded into the target Protocol Buffer.
+/// - Buffering the request body fails.
+///
+/// See [`ProtoBufRejection`] for more details.
+///
+/// # Extractor example
+///
+/// ```rust,no_run
+/// use axum::{routing::post, Router};
+/// use axum_extra::protobuf::ProtoBuf;
+///
+/// #[derive(prost::Message)]
+/// struct CreateUser {
+///     #[prost(string, tag="1")]
+///     email: prost::alloc::string::String,
+///     #[prost(string, tag="2")]
+///     password: prost::alloc::string::String,
```

Comment:
I don't suppose the full qualification is necessary.

```suggestion
///     email: String,
///     #[prost(string, tag="2")]
///     password: String,
```